pdns/rcpgenerator.cc

```diff
@@ -50,6 +50,29 @@ void RecordTextReader::xfr48BitInt(uint64_t &val)
     throw RecordTextException("Overflow reading 48 bit integer from record content"); // fixme improve
 }
 
+void RecordTextReader::xfrNodeOrLocatorID(NodeOrLocatorID& val) {
+  skipSpaces();
+  size_t len;
+  for(len=0;
+      d_pos+len < d_string.length() && (isxdigit(d_string.at(d_pos+len)) || d_string.at(d_pos+len) == ':');
+      len++);
+
+  if(len != 19) {
+    throw RecordTextException("while parsing colon-delimited 64-bit field: length is incorrect, "+ std::to_string(len) + " instead of 19");
+  }
+
+  // HACK HACK HACK, parse as v6, and then strip the final 64 zero bytes
+  struct in6_addr tmpbuf;
+  string to_parse = d_string.substr(d_pos, len) + "::";
```

Comment:
Or, append `:0:0:0:0` here instead of `::` so a wrong colon count above is caught by `inet_pton` below.